packages/compass-editor/src/types.d.ts

```diff
@@ -0,0 +1,58 @@
+import type AceBuilds from 'ace-builds';
+import type { Ace } from 'ace-builds';
+
+type GlobalAce = {
+  EditSession: typeof AceBuilds['EditSession'];
+  Range: typeof AceBuilds['Range'];
+  UndoManager: typeof AceBuilds['UndoManager'];
+  VirtualRenderer: typeof AceBuilds['VirtualRenderer'];
+  config: typeof AceBuilds['config'];
+  createEditSession: typeof AceBuilds['createEditSession'];
+  edit: typeof AceBuilds['edit'];
+  require: typeof AceBuilds['require'];
+  version: typeof AceBuilds['version'];
+  acequire: typeof AceBuilds['require'];
+  define: (
+    module: string,
+    depdendencies: string[],
+    moduleFn: (
+      acequire: typeof AceBuilds['require'],
+      exports: any,
+      module: any
+    ) => void
+  ) => void;
+};
+
+declare global {
+  declare const ace: GlobalAce;
+}
+
+export interface HighlightRules {
+  normalizeRules(): void;
+  createKeywordMapper(
+    map: Record<string, string>,
+    defaultToken: string
+  ): (...args: any[]) => any;
+  $rules: Record<
+    string,
+    {
+      regex: string | RegExp;
+      token: string | string[] | ((...args: any[]) => any);
+      next?: string;
+    }[]
+  >;
+}
+
+export interface WorkerClient {
+  attachToDocument(doc: Document): void;
+}
+
+export interface AceMode {
```

Comment:
do we need to export this and `WorkerClient`?

Reply:
WorkerClient I can remove, AceMode type we use in `src/ace/mode.ts`

Reply:
Done in https://github.com/mongodb-js/compass/pull/3589/commits/562c7b1287336e64d71281a249dc970e097eee28

Reply:
Oh, yeah, no, good question, so no, nothing from here is exported outside the package, it is only so I can import them in ace/mode code